Source/WebCore/html/HTMLTextFormControlElement.cpp

```diff
@@ -853,8 +853,9 @@ void HTMLTextFormControlElement::adjustInnerTextStyle(const RenderStyle& parentS
         }
     }
 
-    if (isDisabledFormControl())
-        textBlockStyle.setColor(RenderTheme::singleton().disabledTextColor(textBlockStyle.visitedDependentColorWithColorFilter(CSSPropertyColor), parentStyle.visitedDependentColorWithColorFilter(CSSPropertyBackgroundColor)));
+    // FIXME: Remove me once we are sure, we get the right result.
+    // if (isDisabledFormControl())
+    //     textBlockStyle.setColor(textBlockStyle.visitedDependentColorWithColorFilter(CSSPropertyColor), parentStyle.visitedDependentColorWithColorFilter(CSSPropertyBackgroundColor));
```

Comment:
I don't know why?

So this is not working yet. 
If I set a color in the document stylesheet as a webdev for `input:disabled` it is picking the right color.
BUT by default it doesn't seem to pick up the color specified by `color-mix` for the `textBlockStyle`. The border and widgets are taking the color. 

```
data:text/html,<!doctype%20html><html><input%20type="text"%20value="ABCDEFGHIJ"%20disabled="">
```


@weinig , Would you have a hint ?

Reply:
I think `color-mix()` currently doesn't support system colors (such as `CanvasText`)

Reply:
Works on latest trunk WebKit for me:

`data:text/html,<span%20style="color:color-mix(in%20hsl,%20CanvasText%2066%,%20Canvas);">fdsdfsdffddsf</span>`

Reply:
yes the `color-mix()` is working when I set it manually, but it doesn't through `html.css` and I'm not sure why.

Reply:
I found out why, https://searchfox.org/wubkat/rev/3de3ef30cefada3014150b0591598345dc7458e3/Source/WebCore/css/parser/CSSParserContext.cpp#57

You need to add `colorMixEnabled = true` there and it should work.

Reply:
🤯 Thanks! Let's try.